dietpi/misc/bannerwrap_fold.awk

```diff
@@ -0,0 +1,98 @@
+
+function make_whitespace_offset(clean_line){
+    ## Generate leading spaces based on global INDENT_TYPE
+    if (INDENT_TYPE == "min"){
+        return sprintf("%*s", INDENT_MIN, "")
+    }
+    lead_spaces = ""
+    if ((INDENT_TYPE == "dash" || INDENT_TYPE == "dashcolon") && match(clean_line, green_dash)){
+        lead_spaces = sprintf("%*s", RSTART + RLENGTH - 2, "")
+    }
+    if ((INDENT_TYPE == "colon" || INDENT_TYPE == "dashcolon") && match(clean_line, green_colon)){
+        lead_spaces = sprintf("%*s", RSTART + RLENGTH - 2, "")
+    }
+    # Useable space check
+    # - "Let's Encrypt cert status" would be squashed otherwise
+    if ( (MAXCOL - length(lead_spaces)) < MIN_USEABLE_SPACE){
+        lead_spaces = sprintf("%*s", INDENT_MIN, "")
+    }
+    return lead_spaces
+}
+
+BEGIN {
+    green_dash = "^[[:space:]]+-[[:space:]]"  # green "bullets"
+    green_colon = "[[:space:]]:[[:space:]]"  # green colon in middle
+    color_regex = "[[:cntrl:]][[0-9;?]*[A-Za-z]"  # all color codes
+    ## Init from CLI, or use defaults here
+    if (MIN_USEABLE_SPACE == ""){MIN_USEABLE_SPACE = 8} ## min space on the right to keep
+    if (MAX_COL == ""){MAX_COL = 30}                    ## Wrap to column number
+    if (INDENT_MIN == ""){INDENT_MIN = 1}               ## min offset seen in dietpi banner
+    if (INDENT_TYPE == ""){INDENT_TYPE = "dashcolon"}   ## Indent modes:
+    ## min - use minimum
+    ## dash - green_dash only
+    ## colon - green_colon only
+    ## dashcolon - match dash first then colon (colon overrides)
+}
+
+{
+    ## ASCII ART: Skip or Hide
+    if (match($0, /^[^a-zA-Z0-9─]+$/)){
+        if (MAXCOL > (RSTART+RLENGTH)){print $0;}
+        next;
+    }
+    ## Green Lines: Truncate to MAXCOL
+    if (match($0, /────/)){
+        new_grn_line = sprintf("%*s", MAXCOL - 2, "")    ## new line: of spaces
+        gsub(/ /, "─", new_grn_line)                     ## new line: change spaces to line char
+        gsub(/─/, "W", $0)  # hack:mawk cant handle /─*/ ## old line: change all line chars to W
+        sub(/W+/, new_grn_line, $0)                      ## old line: change longest W to new line
+        print; next;
+    }
+    ## Begin word-wrapping
+    ## - Wrap based on stripped colour position
+    ## - Indent to either first green dash, or first green colon
+    ##   based on INDENT_TYPE arg
+    ## - If useable space after indenting is too small, or a long
+    ##   word such as an URL cannot be split then indent to INDENT_MIN
+    clean_line = $0
+    gsub(color_regex, "", clean_line)      ## Line without color
+    gsub(/[[:space:]]+/, " ", clean_line)  ## "dietpi-config" misaligned without this
+
+    ## tokenize coloured line into "words", stripped line into "stripped_words"
+    wn = split($0, words, /[[:space:]]+/)
+    sn = split(clean_line, stripped_words, /[[:space:]]+/)
+
+    if (sn > 0 && wn != sn){
+        print "[ERROR] Word ordering does not match stripped word ordering"
+        exit 1
+    }
```

Comment:
Maybe this should fail with a warning or silently, otherwise I wonder if anything will be printed at all